dawn/src/dawn/CodeGen/CXXNaive/CXXNaiveCodeGen.cpp

```diff
@@ -447,53 +447,62 @@ void CXXNaiveCodeGen::generateStencilClasses(
 
                 auto const& extents = iir::extent_cast<iir::CartesianExtent const&>(
                     stage.getExtents().horizontalExtent());
-                auto doMethodGenerator = [&]() {
-                  // Check if we need to execute this statement:
-
-                  // Generate Do-Method
-                  for(const auto& doMethodPtr : stage.getChildren()) {
-                    const iir::DoMethod& doMethod = *doMethodPtr;
-                    if(!doMethod.getInterval().overlaps(interval))
-                      continue;
-                    for(const auto& stmt : doMethod.getAST().getStatements()) {
-                      stmt->accept(stencilBodyCXXVisitor);
-                      stencilRunMethod << stencilBodyCXXVisitor.getCodeAndResetStream();
+
+                // Check if we need to execute this statement:
+                bool hasBody = false;
```

Comment:
`hasBody` is a misleading name, all the stages do have a body. The question is if they have overlapping do-method intervals